### PR TITLE
Us 24

### LINKED ISSUE #59: User Story 24, User can manipulate quantities in their cart
```
As a visitor or registered user
When I have items in my cart
And I visit my cart
Next to each item in my cart
I see a button or link to remove that item from my cart
- clicking this button will remove the item but not other items

I see a button or link to increment the count of items I want to purchase
- I cannot increment the count beyond the merchant's inventory size

I see a button or link to decrement the count of items I want to purchase
- If I decrement the count to 0 the item is immediately removed from my cart
```

DO THESE STORIES FIRST:
- User Story 21, User views their cart show page with items in the cart

### Checklist:

- [x] base tests are written
- [x] base code is written to pass tests
- [x] edge case ("sad path") tests written (if required)
- [x] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #249

child of #274

depends on #247

### PR DESCRIPTION
**Pull request by WHomer**
What does this PR do?
Resolves #59

### CODE REVIEW

File: app/controllers/carts_controller.rb [hunk 1]
```diff
@@ -6,6 +6,22 @@ def show
     @cart_price_total = cart.total_price
   end
 
+  def update
+    request = params[:request]
+    item_id = params[:id]
+    cart = session[:cart]
+    if request == "remove_item"
+      cart.delete(item_id)
+    elsif request == "add_one"
```
Comment (zachlp2016) [inline app/controllers/carts_controller.rb:15]: Very gouda

File: spec/features/carts/show_spec.rb [hunk 2]
```diff
@@ -51,22 +37,121 @@
...
+          find "img[src='https://kaaskraam.com/wp-content/uploads/2018/02/Gouda-Belegen.jpg']"
+        end
+
+        click_link "Clear Cart"
+
+        expect(page).to_not have_content(@item_1.name)
+        expect(page).to_not have_content(@item_2.name)
+        expect(page).to_not have_content(@item_3.name)
+        expect(page).to have_content("(0)")
+      end
```
Comment (zachlp2016) [inline spec/features/carts/show_spec.rb:109]: Very thorough testing. Grate job!

**Review by zachlp2016 — APPROVED** [review]
